pkg/telemetryquerylanguage/tql/parser.go

```diff
@@ -75,7 +115,7 @@ type Field struct {
 // invocation and the condition to match telemetry for invoking the function.
 type Query struct {
 	Function  ExprFunc
-	Condition CondFunc
```

Comment:
Should we rename `Condition` in this struct to something else?  We are redefining what a condition means in the grammar so the returned struct might benefit from a clearer name.  I also can't think of a better name, but I don't love this being exposed as "Condition" while internally there is this other thing called "Condition" that is know just a building block of the where clause.  Or maybe thats ok bc the user doesn't care about our internals and just wants to be able to do 

```
if statement.Condition(ctx) {
}
```